pkgs/by-name/bs/bsc/package.nix

```diff
@@ -20,17 +20,22 @@ stdenv.mkDerivation (finalAttrs: {
 
   buildInputs = lib.optional stdenv.hostPlatform.isDarwin llvmPackages.openmp;
 
+  postPatch = lib.optional (!stdenv.hostPlatform.isx86) ''
+    sed -i '/-mavx2/d' makefile
```

Comment:
looking at the makefile this or a patch is probably the best way to remove that item.
I'd say substituteInPlace with the replace fail flag would be more resilient.